lesson_07/homework_07.py

```diff
@@ -3,69 +3,237 @@
+    return a + b
+
+
+sum_ints(8.55, 19)  # 8.55 + 19 = 27.55
+
 
 # task 3
 """  Написати функцію, яка розрахує середнє арифметичне списку чисел.
 """
 
+def arithmetic_mean(nums):
+    """
+    This function prints an arithmetic mean
+    of any count of numbers in list.
+    :param nums: list of ints/floats
+    """
+    arithmetic = sum(nums) / len(nums)
+    print(arithmetic)
+
+
+list_of_nums = [10, 40]
+arithmetic_mean(list_of_nums)  # 25.0
+
+
 # task 4
 """  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
 """
 
+def reverse_string(string):
+    """
+    This function reverses any string.
+    :param string: string
+    :return: reversed string
+    """
+    return string[::-1]
+
+
+my_str = "This is string to reverse!"
+print(reverse_string(my_str))  # !esrever ot gnirts si sihT
+
 # task 5
 """  Написати функцію, яка приймає список слів та повертає найдовше слово у списку.
 """
 
+
+def longest_word(words: list):
+    """
+    This function prints as result of the longest string in list.
+    :param words: list of strings
+    """
+    result = max(words, key=len)
+    print(result)
+
+
+list_of_words = ['samsung', 'apple', 'xaomi']
+longest_word(list_of_words)  # samsung
+
 # task 6
 """  Написати функцію, яка приймає два рядки та повертає індекс першого входження другого рядка
 у перший рядок, якщо другий рядок є підрядком першого рядка, та -1, якщо другий рядок
 не є підрядком першого рядка."""
+
+
 def find_substring(str1, str2):
+    """
+    This function returns index of first letter of second string in first one.
+    The function return '-1' if second sting doesn't in first one.
+    :param str1: str
+    :param str2: str
+    :return: index str2 in str1 // -1
+    """
+    if str2 in str1:
+        return str1.index(str2)
+    else:
+        return -1
 
-    return -1
 
-str1 = "Hello, world!"
-str2 = "world"
-print(find_substring(str1, str2)) # поверне 7
+str_1 = "Hello, world!"
+str_2 = "world"
+print(find_substring(str_1, str_2))  # 7
 
-str1 = "The quick brown fox jumps over the lazy dog"
-str2 = "cat"
-print(find_substring(str1, str2)) # поверне -1
+str_1 = "The quick brown fox jumps over the lazy dog"
+str_2 = "cat"
+print(find_substring(str_1, str_2))  # -1
+
+
+# task 7-10
+"""  Оберіть будь-які 4 таски з попередніх домашніх робіт та
+перетворіть їх у 4 функції, що отримують значення та повертають результат.
+Обоязково документуйте функції та дайте зрозумілі імена змінним.
+"""
 
 # task 7
+# Задача з використанням циклу for та continue. Задано список фруктів 'fruits'
+# потрібно вивести на екран всі елементи списку, окрім "orange"
+
+def exclude(obj, excl):
+    """
+    This function can exclude item from list/set/tuple/dict and prints result.
+    Your original variable won't be changed.
+    :param obj: list/set/tuple/dict
+    :param excl: str
+    :return: Your variable without excluding item.
+    """
+    res = [print(item) for item in obj if item != excl]
```

Comment:
```suggestion
    res = [item for item in obj if item != excl]
```
помлка, оскільки print(item) повертає None